calc/policy_sorter.go

```diff
@@ -91,6 +96,35 @@ func (p PolKV) String() string {
 	return fmt.Sprintf("%s(%s)", p.Key.Name, orderStr)
 }
 
+func (p PolKV) governsType(wanted string) bool {
+	// Back-compatibility: no Types means Ingress and Egress.
+	if len(p.Value.Types) == 0 {
+		return true
+	}
+	for _, t := range p.Value.Types {
+		if strings.EqualFold(t, wanted) {
+			return true
+		}
+	}
+	return false
+}
+
+func (p PolKV) GovernsIngress() bool {
+	if p.Ingress == nil {
+		governsIngress := p.governsType("ingress")
+		p.Ingress = &governsIngress
```

Comment:
Excellent point.

As I was going to sleep last night, I was thinking I'd change it to (p *PolKV) for efficiency, but I didn't spot that it was actually bugged as is.